src/registry.ts

```diff
@@ -47,7 +55,11 @@ export default class Registry implements RegistryAccessor {
   }
 
   registration(specifier: string): FactoryDefinition<any> {
-    return this._registrations[specifier];
+    let registration = this._registrations[specifier];
+    if (this._fallback && registration === undefined) {
```

Comment:
if we reverse the order it should be just slightly more efficient: `if (registration === undefined && this._fallback) {`